internal/gengapic/metadata_test.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gengapic
+
+import (
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/genproto/googleapis/gapic/metadata"
+	metadatapb "google.golang.org/genproto/googleapis/gapic/metadata"
+)
+
+func TestAddMetadataServiceForTransport(t *testing.T) {
+	for _, tst := range []struct {
+		service, lib string
+		init, want   *metadatapb.GapicMetadata
+	}{
+		{
+			service: "LibraryService",
+			lib:     "LibraryServiceClient",
+			init: &metadatapb.GapicMetadata{
+				Services: make(map[string]*metadatapb.GapicMetadata_ServiceForTransport),
+			},
+			want: &metadatapb.GapicMetadata{
```

Comment:
minor: I'm not expert on Go by any means, but I have to say this has way too many indentations =) Is it ok/idiomatic for go? Maybe it can be reduced (number of indentations) by introducing intermediate variables somewhere?

Reply:
lol.... This is totally idiomatic for Go. I guess I've gotten used to this pattern and actually kinda like it.

Reply:
I understand where you are coming from, and this specific case deep indention is normal. This is a Table Driven Test (https://github.com/golang/go/wiki/TableDrivenTests), but one with the "row" type declared in the `range` statement. What makes this worse is the many nested messages we have to initialize :)